streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/ExecutorManager.scala

```diff
@@ -82,24 +86,39 @@ private[appmaster] class ExecutorManager (
     case StartExecutorSystemTimeout =>
       taskManager ! StartExecutorsTimeOut
 
-    case RegisterExecutor(executor, executorId, resource, workerId) =>
+    case RegisterExecutor(executor, executorId, resource, worker) =>
       LOG.info(s"executor $executorId has been launched")
       //watch for executor termination
       context.watch(executor)
-      taskManager ! ExecutorStarted(executor, executorId, resource, workerId)
+      executors += executorId -> ExecutorInfo(executorId, executor, worker)
+      taskManager ! ExecutorStarted(executorId, resource, worker.workerId)
 
     case BroadCast(msg) =>
-      LOG.info(s"broadcasting ${msg.getClass.getName}")
-      context.children.foreach(_ ! msg)
+      LOG.info(s"Broadcast ${msg.getClass.getSimpleName} to all executors")
+      context.children.foreach(_ forward  msg)
+
+    case UniCast(executorId, msg) =>
+      LOG.info(s"Unicast ${msg.getClass.getSimpleName} to executor $executorId")
+      val executor = executors.get(executorId)
+      executor.foreach(_.executor forward msg)
 
     case GetExecutorPathList =>
       sender ! context.children.map(_.path).toList
+
+    case ExecutorResourceUsageSummary(resources) =>
```

Comment:
what is this message used for? Looks unclear to me. Better add some comment on the ```case``` clause.